README.md

```diff
@@ -20,14 +20,28 @@ However, it is a good idea to consult name changes with other people - use pull
 ## Gradle
 Yarn uses Gradle to provide a number of utility tasks for working with the mappings.
 
+### `yarn`
+[`setupYarn`](#setupYarn) and download and launch the latest version of [Enigma](https://github.com/FabricMC/Enigma) automatically configured to use the merged jar and the mappings.
+
+Compared to launching Enigma externally, the gradle task adds a name guesser plugin that automatically map enums and a few constant field names.
+
+### `build`
+[`compressTiny`](#compressTiny) and [`buildEnigma`](#buildEnigma).
+
+### `mapNamedJar`
+Builds a deobfuscated jar with yarn mappings and automapped fields (enums, etc.). Unmapped names will be filled with [intermediary](https://github.com/FabricMC/Intermediary) names.
+
 ### `download`
 Downloads the client and server Minecraft jars for the current Minecraft version to `.gradle/minecraft`
 
 ### `mergeJars`
 Merges the client and server jars into one merged jar, located at `VERSION-merged.jar` in the mappings directory where `VERSION` is the current Minecraft version.
 
 ### `setupYarn`
-`download` and `mergeJars`
+[`download`](#download) and [`mergeJars`](#mergeJars)
 
-### `yarn`
-`setupYarn` and download and launch the latest version of Enigma automatically configured to use the merged jar and the mappings.
+### `buildEnigma`
+Packages enigma mappings into a zip archive.
+
+### `compressTiny`
```

Comment:
not sure this is needed, the whole idea is to just run gradle build

Reply:
Build produces 2 artifacts, so I am explaining one by one

Reply:
Doesnt matter, there should be no reason for anyone to run compressTiny, just use build